inc/plugin.class.php

```diff
@@ -500,25 +500,31 @@ function listPlugins() {
                case self::ANEW :
                case self::NOTINSTALLED :
                case self::NOTUPDATED :
-                  echo "<td>";
+                  $out = '';
+                  $do_install = true;
                   if (function_exists("plugin_".$plug['directory']."_install")
                       && function_exists("plugin_".$plug['directory']."_check_config")) {
 
                      $function   = 'plugin_' . $plug['directory'] . '_check_prerequisites';
-                     $do_install = true;
                      if (function_exists($function)) {
+                        ob_start();
                         $do_install = $function();
+                        if (!$do_install) {
+                           $out .= '<span class="error">' . ob_get_contents() . '</span>';
+                        }
+                        ob_end_clean();
```

Comment:
I do not like that at all... But since plugin (and core also) used to "echo" directly, there is no real alternative :'(